e2e/content-services/pages/permissions.page.ts

```diff
@@ -15,29 +15,97 @@
     addPermissionButton = TestElement.byCss("button[data-automation-id='adf-add-permission-button']");
     addPermissionDialog = element(by.css('adf-add-permission-dialog'));
     searchUserInput = element(by.id('searchInput'));
     searchResults = element(by.css('#adf-add-permission-authority-results #adf-search-results-content'));
-    addButton = element(by.id('add-permission-dialog-confirm-button'));
+    addButton = element(by.css('[data-automation-id="add-permission-dialog-confirm-button"]'));
     permissionInheritedButton = element.all(by.css('.app-inherit_permission_button button')).first();
     noPermissions = element(by.id('adf-no-permissions-template'));
     deletePermissionButton = element(by.css(`button[data-automation-id='adf-delete-permission-button']`));
     permissionDisplayContainer = element(by.id('adf-permission-display-container'));
     closeButton = TestElement.byCss('#add-permission-dialog-close-button');
 
     async changePermission(name: string, role: string): Promise<void> {
-        await this.addPermissionsDialog.clickRoleDropdownByUserOrGroupName(name);
-        await this.addPermissionsDialog.selectOption(role);
+        await this.clickRoleDropdownByUserOrGroupName(name);
+        await new DropdownPage().selectOption(role);
+        await this.dataTableComponentPage.checkRowByContentIsNotSelected(name);
+    }
+
+    async checkUserIsAdded(id: string) {
+        const userOrGroupName = TestElement.byCss('div[data-automation-id="' + id + '"]');
+        await userOrGroupName.waitPresent();
+    }
+
+    async getRoleCellValue(username: string): Promise<string> {
+        const locator = this.dataTableComponentPage.getCellByRowContentAndColumn('Users and Groups', username, 'Role');
+        return BrowserActions.getText(locator);
+    }
+
+    async clickRoleDropdownByUserOrGroupName(name: string): Promise<void> {
+        const row = this.dataTableComponentPage.getRow('Users and Groups', name);
+        await row.click();
+        await BrowserActions.click(row.element(by.css('[id="adf-select-role-permission"] .mat-select-trigger')));
+    }
+
+    async clickDeletePermissionButton(username: string): Promise<void> {
+        const userOrGroupName = TestElement.byCss(`[data-automation-id="adf-delete-permission-button-${username}"]`);
+        await userOrGroupName.waitPresent();
+        await userOrGroupName.click();
+
+    }
+
+    async checkUserIsDeleted(username: string): Promise<void> {
+        const userOrGroupName = TestElement.byCss('div[data-automation-id="' + username + '"]');
+        await userOrGroupName.waitNotPresent();
+    }
+
+    async noPermissionContent(): Promise<string> {
+        const noPermission = TestElement.byCss('.adf-no-permission__template--text');
+        return noPermission.getText();
+    }
+
+    async waitVisible(): Promise<void> {
+        await this.rootElement.waitPresent();
+    }
+
+    async isErrored(): Promise<boolean> {
```

Comment:
looking at the element name and selector, "emptyElement", ".adf-no-permission__template", I don't understand the name of this method "isErrored", or the name of the following method: "waitForError"
please find better / more relevant naming.
If ".adf-no-permission__template" really is an error, then the element should not be called "emptyElement"